src/audio/copier/copier.c

```diff
@@ -990,6 +987,49 @@ static int copier_get_hw_params(struct comp_dev *dev, struct sof_ipc_stream_para
 	return dai_common_get_hw_params(dd, dev, params, dir);
 }
 
+static int copier_bind(struct processing_module *mod, void *data)
+{
+	const struct ipc4_module_bind_unbind *const bu = (struct ipc4_module_bind_unbind *)data;
+	struct copier_data *cd = module_get_private_data(mod);
+	struct comp_dev *dev = mod->dev;
+	struct comp_buffer *buffer;
+	uint32_t src_id, sink_id;
+	struct list_item *list;
+
+	src_id = IPC4_COMP_ID(bu->primary.r.module_id, bu->primary.r.instance_id);
```

Comment:
I think the whole function could be simplified. 
```suggestion
	uint32_t src_queue_id = bu->extension.r.src_queue;
	uint32_t dst_id = IPC4_COMP_ID(bu->extension.r.dst_module_id,
				       bu->extension.r.dst_instance_id);
	if (dev->ipc_config.id == dst_id)
		return 0;

	/* Update corresponding sink buffer format */
	list_for_item(sink_list, &dev->bsink_list) {
		int sink_id;

		sink = container_of(sink_list, struct comp_buffer, source_list);
		sink_id = IPC4_SRC_QUEUE_ID(buf_get_id(sink));
		if (sink_id == src_queue_id) {
			ipc4_update_buffer_format(sink, &cd->out_fmt[sink_id]);
			return 0;
		}
	}

	comp_err(dev, "No sink buffer found for src_queue=%d", src_queue_id);
	return -ENODEV;
```